packages/collection-model/index.d.ts

```diff
@@ -1,7 +1,18 @@
 import type toNS from 'mongodb-ns';
 import type { DataService } from 'mongodb-data-service';
 
-type Namespace = ReturnType<typeof toNS>;
+type CollectionMetadata = {
+  namespace: string;
+  isReadonly: boolean;
+  isTimeSeries: boolean;
+  isClustered: boolean;
+  isFLE: boolean;
+  isSearchIndexesSupported: boolean;
+  sourceName?: string;
```

Comment:
I'm wondering if this is maybe a good place to document what these things mean. I'm realising I can kinda infer what the is* ones mean, but for sourceName, sourceReadonly, sourceViewon, sourcePipeline I have no idea.

(I know this is not new in your PR, just a thought I immediately had while reading the code.)

Reply:
Yes, good idea, I'll do that! We did some good job recently documenting `setupStore` options for refactored plugins and so it would make sense to add some descriptions here too

Reply:
Added description in https://github.com/mongodb-js/compass/pull/4858/commits/b58813da58faa19431b37f8b60f322d232ee7f19